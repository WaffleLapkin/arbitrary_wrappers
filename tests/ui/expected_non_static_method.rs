#![feature(arbitrary_self_types)]

// TODO: make the error cleaner

use arbitrary_wrappers::use_ast;

fn main() {
    assert_eq!(Wrapper(0, Type).method(), 0);
}

struct Wrapper<T>(i32, T);

struct Type;

#[use_ast(Type)]
impl Wrapper<Type> {
    fn method() -> i32 {
        0
    }
}

impl<T> core::ops::Deref for Wrapper<T> {
    type Target = T;

    fn deref(&self) -> &Self::Target {
        &self.1
    }
}
