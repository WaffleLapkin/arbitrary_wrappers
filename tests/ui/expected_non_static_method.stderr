error: custom attribute panicked
  --> $DIR/expected_non_static_method.rs:15:1
   |
15 | #[use_ast]
   | ^^^^^^^^^^
   |
   = help: message: Expected non-static method, found method with 0 arguments. Consider moving this function to another impl block.
